include/aws/http/connection.h

```diff
@@ -56,6 +56,13 @@ typedef void(
 typedef void(
     aws_http_on_client_connection_shutdown_fn)(struct aws_http_connection *connection, int error_code, void *user_data);
 
+/**
+ * Invoked when the HTTP/2 connection settings ack frame received.
+ * It can be not invoked if we never received the settings ack frame for that settings.
+ * This is always invoked on connetion's event-loop thread.
+ */
+typedef int(aws_http2_on_settings_ack_received)(void *user_data);
```

Comment:
Suggestion:
`typedef void (aws_http2_on_change_settings_complete)(struct aws_http_connection *connection, int error_code, void *user_data)`

Removed `int` return type. Most callbacks, and especially "completion" callbacks, shouldn't affect the thing calling them. I know that *most* of the stream callbacks have an `int` return type, but these are a special case. A long series of stream callbacks that all must succeed : an exception from the user while receiving headers, generally means the user is not in a position to receive the body either. The stream's completion callback, which fires last, and is guaranteed to fire, has a `void` return type.

Also, I would ALWAYS fire the callback. It's way easier to hook these up in bindings when we know they will fire exactly once: we can give the user some "future" type that they can wait on. If we never receive the ACK, we can fire it with an `AWS_ERROR_HTTP_CONNECTION_CLOSED` error_code when tearing things down and cleaning up the pending_settings list.